testtools/files.go

```diff
@@ -129,3 +139,232 @@ func CheckFiles(t *testing.T, dir string, files []FileSpec) {
 		}
 	}
 }
+
+type TestGazelleGenerationArgs struct {
+	// name is the name of the test.
+	Name string
+	// testDataPath is the workspace relative path to the test data directory.
+	TestDataPath string
+	// gazelleBinaryDir is the workspace relative path to the location of the gazelle binary
+	// we want to test.
+	GazelleBinaryDir string
```

Comment:
Can you collapse `GazelleBinaryDir` and `GazelleBinaryName` into a single attribute?